### DISCUSSION TITLE
Syntax highlighting, from running github-markup locally?

### DISCUSSION THREAD

**Discussion by deftclaw** [github-linguist/linguist#6716] [Q&A] [answered]
![generated](https://github.com/github-linguist/linguist/assets/57646863/171b6cbc-29bd-4219-a367-d95506a2ec02)
Got github/markup converting *.md to *.html nicely, using custom CSS to make swanky notes --> documentation editor. That part is great. I wanted to add syntax-highlighting for code blocks and the github-markup repository [seemed to indicate](https://github.com/github/markup#:~:text=Syntax%20highlighting%20is%20performed%20on%20code%20blocks.%20See%20github/linguist%20for%20more%20information%20about%20syntax%20highlighting.) that this was where to look.

All the docs here seem to just be listing how much of which languages are in a repo. That's great in it's own right, but it does not seem to be what the github/markup readme implied.

How can I utilize this tool to perform syntax highlighting with github/markup locally?
From the README the only use of this tool I can grasp is stats of included code, but there seems to be plenty of discussion about highlighting going on...

cannot find 'syntax' or 'highlig' in [the docs](https://www.rubydoc.info/gems/github-linguist/7.28.0/Linguist)  
so if you're going to 'RTFM' me, please add what manual I ought FR?

Please help... am I being as stupid as I feel?
Thanks in advance!
```ruby
require 'github/markup'
require 'linguist'

page_name  = file.gsub(/\.md/, '.html')
conversion = GitHub::Markup.render(file, File.read(file))
```

```md
# This is to test code highlighting

```ruby
def ducktales
  puts "woohoo!"
end
``\`

```

**Comment by lildude** [comment, marked as answer]
> How can I utilize this tool to perform syntax highlighting with github/markup locally?

You can't 😁 Linguist doesn't do the syntax highlighting, it only collects the grammars which are passed to the syntax highlighting engine (that's the linguist-grammars.tar.gz file attached to each [release](https://github.com/github-linguist/linguist/releases)).

This highlighting engine GitHub uses is not open source and is not available for use outside GitHub but if you want local syntax highlighting you'll need to look at something else like [rouge](https://github.com/rouge-ruby/rouge) which [Jekyll](https://jekyllrb.com/docs/liquid/tags/) uses (Ruby) or [chroma](https://github.com/alecthomas/chroma) which [Hugo](https://gohugo.io/content-management/syntax-highlighting/) uses (Go). Both are static site generators that convert markdown to HTML, potentially saving you from reinventing the wheel.

There are others but that should get you going.

**Reply by deftclaw** [reply]
Thanks so much! I'll checkout rouge
